src/events/AnalyticsEvents.ts

```diff
@@ -76,14 +76,8 @@ export interface IChangeAnalyticsCustomDataEventArgs extends IChangeableAnalytic
 export interface IChangeableAnalyticsMetaObject {
   /**
    * A metadata key-value pair to send along with the usage analytics event.
-   *
-   * **Notes:**
-   * > * A metadata key must contain only alphanumeric characters and underscores (the Coveo Usage Analytics REST
-   * > service automatically converts white spaces to underscores and uppercase characters to lowercase characters in key
-   * > names).
-   * > * A metadata value must be a simple string (no other type is allowed).
    */
-  [name: string]: string;
+  [name: string]: any;
```

Comment:
UA now accept objects as meta values, not just string. They parse into strings on their end. Maybe I should put a comment talking about that instead? @fbeaudoincoveo

Reply:
Ah! I thought this was already the case. In the UA Write API Swagger, the `customData` description currently is:

> The user defined dimensions and their values. Keys can only contain alphanumeric or underscore characters. Whitespaces in keys are converted to underscores. Uppercase characters in keys are converted to lowercase characters. The value could be any valid JSON, but it is handled as string over Coveo usage analytics. It is highly recommended that you create your custom dimension before adding customData

Interestingly, the argument received by `changeAnalyticsCustomData` event handlers has a more restrictive definition of a `metaObject` than the UA logging top-level functions (in the latter, the `meta` argument accepts `any` type).

We have a some discrepancies about metadata in the documentation (both in the reference and on docs.coveo.com). When you close this Jira, please select the "requires public documentation" box so I'm reminded to deal with those :)

Reply:
Yes, reverting to the original documentation would be fine.